library/std/src/io/mod.rs

```diff
@@ -1957,6 +1957,32 @@ pub trait Seek {
     fn stream_position(&mut self) -> Result<u64> {
         self.seek(SeekFrom::Current(0))
     }
+
+    /// Seeks relative to the current position.
+    ///
+    /// This is equivalent to `self.seek(SeekFrom::Current(offset))`.
```

Comment:
The documentation could be improved.

It should say that the default implementation is equivalent to that but doesn't return the new position which can allow some implementations such as `BufWriter` (+ link to bufwriter trait impl docs) to perform more efficient seeks.